crates/cxx-qt-lib-extras/src/assertion_utils.h

```diff
@@ -14,27 +15,39 @@
 namespace rust::cxxqtlib1::assertion_utils {
 template<typename iter>
 constexpr static ::std::size_t
-calc_align_size(const iter rbegin,
-                const iter rend,
+calc_align_size(const iter begin,
+                const iter end,
                 const ::std::size_t actual_alignment)
 {
   ::std::size_t rows = 0;
-  ::std::size_t accum = 0;
 
-  for (auto it = rbegin; it != rend; ++it) {
-    assert(*it <= actual_alignment);
+  ::std::size_t row_accum = 0;
+  ::std::size_t max_size_in_row = 0;
 
-    if (it + 1 != rend) {
-      if (accum + *it == actual_alignment) {
-        accum += *it;
+  for (auto it = begin; it != end; ++it) {
+    auto val = *it;
+    assert(val <= actual_alignment);
+
+    if (val > max_size_in_row)
+      max_size_in_row = val;
+
+    if ((row_accum + val) % max_size_in_row == 0) {
+      row_accum += val;
+    } else {
+      auto pad = max_size_in_row - ((row_accum + val) % max_size_in_row);
+
+      if (row_accum + pad + val <= actual_alignment) {
+        row_accum += (pad + val);
       } else {
-        if (accum + *it + *(it + 1) <= actual_alignment)
-          accum += *it;
-        else
-          accum = actual_alignment;
+        ++rows;
+        row_accum = actual_alignment;
```

Comment:
shouldn't this be `row_accum = val`? As otherwise we're immediately filling the next row entirely, no matter how large `val` is?